src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2281,6 +2296,16 @@ private void cleanupStreams(
             sessions.values().forEach(s -> s.cleanup(traceId, authorization));
         }
 
+        private void doEncodeCloseIfNecessary(
+            long traceId,
+            long authorization)
+        {
+            if (!AmqpState.replyClosed(state))
+            {
+                doEncodeClose(traceId, authorization, null, null);
+            }
```

Comment:
This doesn't seem quite right because generally we would expect the state guard to be flipped by the code inside the if check, such that calling it a second time would be a no-op, but that isn't the case here because the state is only flipped when we end the stream.

Perhaps we need a combined `doEncodeCloseAndEndIfNecessary` which would look like above but with `doNetworkEnd(traceId, authorization);` inside the if check, guaranteeing that the state will be flipped as needed and a subsequent call to `doEncodeCloseAndEndIfNecessary` would be a no-op?